cmd/oras/root/push.go

```diff
@@ -182,6 +192,22 @@ func runPush(cmd *cobra.Command, opts *pushOptions) error {
 		}
 		desc.Annotations = packOpts.ConfigAnnotations
 		packOpts.ConfigDescriptor = &desc
+	} else if opts.Platform.Platform != nil {
+		blob, err := json.Marshal(opts.Platform.Platform)
+		if err != nil {
+			return err
+		}
+		mediaType := oras.MediaTypeUnknownConfig
+		if opts.Flag == option.ImageSpecV1_0 && opts.artifactType != "" {
+			mediaType = opts.artifactType
+		}
+		desc := content.NewDescriptorFromBytes(mediaType, blob)
```

Comment:
```suggestion
		mediaType := opts.artifactType
		if mediaType == "" {
			mediaType = opts.artifactType
		}
		desc := content.NewDescriptorFromBytes(mediaType, blob)
```

Reply:
Manually made this change because oras.MediaTypeUnknownConfig.

Reply:
With this change, the e2e test fails because `oras push --oci-layout /tmp/ginkgo1771247066:e2e --artifact-platform darwin/arm64 foobar/bar -v`

Results in:
```
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.unknown.artifact.v1",
  "config": {
    "mediaType": "application/vnd.unknown.artifact.v1",
    "digest": "sha256:e94c0ba80a1157ffab5b5c6656fffc089c6446c7ed0604f3382910d1ef7dd40d",
    "size": 38
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:d1bc8d3ba4afc7e109612cb73acbdddac052c93025aa1f82942edabb7deb82a1",
      "size": 5,
      "annotations": {
        "org.opencontainers.image.title": "foobar/bar"
      }
    }
  ],
  "annotations": {
    "org.opencontainers.image.created": "2024-10-15T16:14:58Z"
  }
}
```

instead of 
```
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.unknown.artifact.v1",
  "config": {
    "mediaType": "application/vnd.unknown.config.v1+json",
    "digest": "sha256:e94c0ba80a1157ffab5b5c6656fffc089c6446c7ed0604f3382910d1ef7dd40d",
    "size": 38
  },
  "layers": [
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar",
      "digest": "sha256:d1bc8d3ba4afc7e109612cb73acbdddac052c93025aa1f82942edabb7deb82a1",
      "size": 5,
      "annotations": {
        "org.opencontainers.image.title": "foobar/bar"
      }
    }
  ],
  "annotations": {
    "org.opencontainers.image.created": "2024-10-15T16:20:21Z"
  }
}
```

Reply:
I think the former is preferred since user doesn't specify `--artifact-spec v1.0` or `--config`?

Reply:
This is a problem tho 
```
  "artifactType": "application/vnd.unknown.artifact.v1",
  "config": {
    "mediaType": "application/vnd.unknown.artifact.v1",
```

Reply:
I think switch-casing on the default value is a better option than setting the value.
```suggestion
                mediaType := opts.artifactType
		if mediaType == "" {
			switch opts.PackVersion {
			case oras.PackManifestVersion1_0:
				mediaType = ocispec.MediaTypeImageConfig
			case oras.PackManifestVersion1_1:
				mediaType = oras.MediaTypeUnknownArtifact
			}
		}
		desc := content.NewDescriptorFromBytes(mediaType, blob)
```

Reply:
> This is a problem tho
> 
> ```
>   "artifactType": "application/vnd.unknown.artifact.v1",
>   "config": {
>     "mediaType": "application/vnd.unknown.artifact.v1",
> ```

This is valid for v1.1 though.

Reply:
It may be valid, but I don't think that was the intention for the user

Reply:
User doesn't specify `--artifact-type` so we cannot know what `config.mediaType` user is expecting.

If user what to specify `artifactType` and `config.mediaType` for v1.1 at the same time, they should use `--artifact-type <artifactType> --config <path>:<config.mediaType>`.

Reply:
```
  "artifactType": "application/vnd.unknown.artifact.v1",
  "config": {
    "mediaType": "application/vnd.unknown.artifact.v1",
```
This LGTM and it is compliant to [v1.1 artifact with config](https://oras.land/docs/concepts/artifact#artifacts-with-config)

Reply:
The reason the test coverage failed was because https://github.com/oras-project/oras/blob/main/cmd/oras/root/push.go#L144 means `opts.artifactType` is never empty here, so I got rid of all the unnecessary logic.